packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/UserTests.kt

```diff
@@ -723,6 +736,53 @@ class UserTests {
         }
     }
 
+    @Test
+    @OptIn(ExperimentalKBsonSerializerApi::class)
+    fun mongoClient_defaultSerializer() = runBlocking<Unit> {
+        val (email, password) = randomEmail() to "123456"
+        val user = runBlocking {
+            createUserAndLogin(email, password)
+        }
+        @OptIn(ExperimentalRealmSerializerApi::class)
+        val client: MongoClient = user.mongoClient(TEST_SERVICE_NAME)
+        assertIs<Int>(client.database(app.clientAppId).collection<CollectionDataType, Int>("CollectionDataType").insertOne(CollectionDataType("object-1")))
+    }
+
+    @Test
+    @OptIn(ExperimentalKBsonSerializerApi::class)
+    fun mongoClient_customSerializer() = runBlocking<Unit> {
+        val (email, password) = randomEmail() to "123456"
+        val user = runBlocking {
+            createUserAndLogin(email, password)
+        }
+        val collectionWithDefaultSerializer =
+            user.mongoClient(TEST_SERVICE_NAME).database(app.clientAppId)
+                .collection<CustomDataType, BsonValue>("SyncDog")
+        assertFailsWithMessage<SerializationException>("Serializer for class 'CustomDataType' is not found.") {
+            collectionWithDefaultSerializer.insertOne(CustomDataType("dog-1"))
+        }
+
+        val collectionWithCustomSerializer =
+            user.mongoClient(TEST_SERVICE_NAME, customEjsonSerializer).database(app.clientAppId)
+                .collection<CustomDataType, CustomIdType>("SyncDog")
+        assertIs<CustomIdType>(collectionWithCustomSerializer.insertOne(CustomDataType("dog-1")))
+    }
+
+    @Test
+    @OptIn(ExperimentalKBsonSerializerApi::class)
+    fun mongoClient_unknownClient() = runBlocking<Unit> {
+        val (email, password) = randomEmail() to "123456"
+        val user = runBlocking {
+            createUserAndLogin(email, password)
+        }
+        val mongoClient = user.mongoClient("UNKNOWN_SERVICE")
+        val collection =
+            mongoClient.database(app.clientAppId).collection<CollectionDataType, ObjectId>("CollectionDataType")
+        assertFailsWithMessage<ServiceException>("service not found: 'UNKNOWN_SERVICE'") {
+            collection.insertOne(CollectionDataType("object-1"))
+        }
+    }
+
```

Comment:
We seem to be missing a test for what happens if you have logged out and try to create a MongoClient?  I guess we could allow it, but throwing seems nicer.

I guess there is also about what happens if you have a handle to a `MongoClient` and then log out. In that case I assume we will just see a Service- or AuthException?